### DISCUSSION TITLE
Some questions about ZIO.sleep and Clock

### DISCUSSION THREAD

**Discussion by jxnu-liguobin** [zio/zio#6878] [Q&A]
I'm upgrading to zio 2.0-rc6 for zio-redis, and I'm a little confused about sleep.

When I use RC6, such a test does not pass.

After some investigation, I found that maybe the reason was that the sleep time was not enough to allow Redis to expire.
> This test uses a local real Redis.
```scala
          for {
            key       <- uuid
            value     <- uuid
            _         <- set(key, value) // redis operate 
            exp       <- pExpire(key, 2000.millis) // redis operate
            response1 <- exists(key) // redis operate
            fiber     <- ZIO.sleep(2050.millis).fork <* TestClock.adjust(2050.millis)
            _         <- fiber.join
            response2 <- exists(key) // redis operate
          } yield assert(exp)(isTrue) && assert(response1)(equalTo(1L)) && assert(response2)(equalTo(0L))
``` 
Actual：
response2 = 1

Expect：
response2 = 0

Wondering if I'm missing something? Thank you.

**Comment by vigoo** [comment]
I guess when you override ‘aspects’ in ‘BaseSpec’ it removes the ‘TestAspect.fibers’ that the base class would add. It’s required for the test clock to work properly.

there was a fix after RC6 that always adds this no matter how you override that method

**Reply by jxnu-liguobin** [reply]
thank you

I'm sure it does cover the aspects.

**Reply by jxnu-liguobin** [reply]
Hi @vigoo Did someone fix it in 2.0.0?

**Reply by vigoo** [reply]
Yes the problem I described with `TestAspect.fibers` is fixed in 2.0.0; even if you override that method, fibers will always be tracked in the tests.

**Reply by jxnu-liguobin** [reply]
> Yes the problem I described with `TestAspect.fibers` is fixed in 2.0.0; even if you override that method, fibers will always be tracked in the tests.

Unfortunately, the above test still does not pass.

**Reply by adamgfraser** [reply]
Can you create a minimized reproducer? What are you doing in the implementation of these Redis operations?

**Reply by jxnu-liguobin** [reply]
There is nothing else to do, as this is just a unit test of Redis commands (testing time-related Redis operations), and I have no way of knowing exactly which piece might be causing the problem. this is a [PR](https://github.com/zio/zio-redis/pull/598/files) upgrade to 2.0.0. maybe there are some wrongs zio usage in the PR?

**Reply by adamgfraser** [reply]
I will take a look.

**Reply by jxnu-liguobin** [reply]
The problem has been solved. I found that the `provideLayer(layer)` made the test fail, I tried to use `withClock(Clock.ClockLive)`, It works fine. Similarly, when I use `provideCustomLayer(layer)`, the test also fails and I use `@@ withLiveEnvironment` or  `withLiveEnvironment(spec)` instead. What might I have gotten wrong about TestAspect?  

I'm very confusing.
